### PR TITLE
DR | Replace moment.js

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > To improve performance of our Decision Review apps, we are replacing the deprecated moment.js library with date-fns
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Platform recommendation
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#80433](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80433)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Using moment to manipulate dates
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
  > Updated tests to use date-fns
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Decision Reviews: NOD, HLR & Supplemental Claims

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by powellkerry — DISMISSED** [review]

File: src/applications/appeals/10182/tests/utils/issues.unit.spec.js
Comment (anniebtran) [inline src/applications/appeals/10182/tests/utils/issues.unit.spec.js:9]: could this be `parseDateWithOffset({ days: -1 })`?

**Review by anniebtran — COMMENTED** [review]
Looks good! Gonna test it out locally and then approve 😄

**Review by anniebtran — DISMISSED** [review]
Left a few comments for some small nits, but not blocking!

File: src/applications/appeals/10182/tests/utils/issues.unit.spec.js
Comment (Mottie) [reply src/applications/appeals/10182/tests/utils/issues.unit.spec.js:9]: Yes! I added `parseDateWithOffset` after this code and thought I went back and updated this pattern. Thanks!

File: src/applications/appeals/shared/tests/utils/dates.unit.spec.js
Comment (Mottie) [reply src/applications/appeals/shared/tests/utils/dates.unit.spec.js:68]: Eileen is checking with CAIA - I won't merge until they approve, or we change the format.

File: src/applications/appeals/shared/utils/dates.js
Comment (Mottie) [reply src/applications/appeals/shared/utils/dates.js:21]: Yes! I "think" altering `date` will still change `newDate` but better to make the code consistent and easy to read.

File: src/applications/appeals/shared/tests/utils/dates.unit.spec.js
Comment (Mottie) [reply src/applications/appeals/shared/tests/utils/dates.unit.spec.js:68]: Ok, we need to remove those ordinal endings

File: src/applications/appeals/996/tests/utils/helpers.unit.spec.js
Comment (Mottie) [inline src/applications/appeals/996/tests/utils/helpers.unit.spec.js:54]: This one stayed as `-1` because the original code was doing this, we missed this change when `parseDate(sub(...))` was added

File: src/applications/appeals/shared/constants.js
Comment (Mottie) [inline src/applications/appeals/shared/constants.js:57]: Changed from `'PPP'` - this was adding the number ordinals.

**Review by anniebtran — APPROVED** [review]
LGTM

**Pull request merged**
